src/ingest-pipeline/airflow/dags/utils.py

```diff
@@ -1392,8 +1392,8 @@ def get_threads_resource(dag_id: str, task_id: Optional[str] = None) -> int:
     if task_id is None:
         task_id = '__default__'
     rec = _lookup_resource_record(dag_id, task_id)
-    assert 'threads' in rec, 'schema should guarantee "threads" is present?'
-    return int(rec['threads'])
+    assert any(['threads' in rec, 'coreuse' in rec]), 'schema should guarantee "threads" or "coreuse" is present?'
+    return int(rec.get('coreuse') if rec.get('coreuse') else rec.get('threads'))
```

Comment:
This can't be right, since 'coreuse' might be 60 and we want to return 60% of the cores on the node.  It needs to get converted to a core count.  Also, it has to never return zero, because I think that would break things- I think it should be ceil(float(rec.get('coreuse'))*0.01*number_of_cores)

Reply:
It is a different approach, currently the calculation is performed on the plugin. But this makes me realize that it will need more coding to make use of this new parameter. I'll refactor it so it is centralized.